src/Parse/Markdown.php

```diff
@@ -31,14 +31,37 @@ class Markdown
     use \Winter\Storm\Support\Traits\Emitter;
 
     /**
-     * @var Winter\Storm\Parse\Parsedown\Parsedown Parsedown instance
+     * @var \Winter\Storm\Parse\Parsedown\Parsedown Parsedown instance
      */
-    protected $parser;
+    protected $parserClass;
+
+    /**
+     * Gets an instance of the parser.
+     *
+     * We return a new instance each time to prevent contamination of clean instances.
+     *
+     * @return Parsedown
+     */
+    protected function getParser()
+    {
+        return new $this->parserClass;
```

Comment:
Why was this changed to no longer cache the parser class instance?

Reply:
It was to prevent cross-contamination of data with subsequent Markdown calls. Each Markdown call will now have a clean instance of the parser to work with.

Reply:
Does the parser contain dangerous state?

Reply:
Not confirmed, but it was better to be safe. I don't expect people would use too many Markdown calls in a single request.